### PR TITLE
[Monitor] Support LA cluster CMK features

### PR DESCRIPTION
**Pull request by jsntcy**
**Description**  
This PR implement the following features:
- LA cluster CRUD
- Link/unlink LA workspace with cluster

**Testing Guide**  

**History Notes**  

`az monitor log-analytics cluster create/update/show/delete/list`
`az monitor log-analytics workspace linked-service create/update/show/delete/list`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/monitor/validators.py [hunk 1]
```diff
@@ -7,6 +7,16 @@
 from azure.cli.core.commands.validators import validate_tags, get_default_location_from_resource_group
 
 
+def make_cluster_creation_as_no_wait_by_default(cmd, namespace):
+    get_default_location_from_resource_group(cmd, namespace)
+
+    # Make cluster creation no wait by default before ADX cluster can be provisioned automatically.
+    # TODO: remove this method and make cluster creation support both wait and no wait
+    #  after ADX cluster can be provisioned automatically.
+    cmd.supports_no_wait = True
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/monitor/validators.py:16]: should we remove this one? If user input `--no-wait False`, will it cause endless loop?

**Comment by yonzhan** [comment]
add to S169

File: src/azure-cli/azure/cli/command_modules/monitor/_client_factory.py [hunk 2]
```diff
@@ -91,3 +91,31 @@ def cf_private_endpoint_connections(cli_ctx, _):
...
+
+def cf_log_analytics_workspace_shared_keys(cli_ctx, _):
+    return _log_analytics_client_factory(cli_ctx).shared_keys
+
+
+def cf_log_analytics_workspace_intelligence_packs(cli_ctx, _):
+    return _log_analytics_client_factory(cli_ctx).intelligence_packs
+
+
+def cf_log_analytics_workspace_linker_service(cli_ctx, _):
```
[the remark this answers is not in the document]
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/monitor/_client_factory.py:116]: yes, typo

---
In reply to: [413632514](https://github.com/Azure/azure-cli/pull/13133#discussion_r413632514) [](ancestors = 413632514)

File: src/azure-cli/azure/cli/command_modules/monitor/validators.py (hunk 1 above)
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/monitor/validators.py:16]: No, we'll not expose --no-wait to users as I didn't add supports_no_wait=True in command definition.

---
In reply to: [413669222](https://github.com/Azure/azure-cli/pull/13133#discussion_r413669222) [](ancestors = 413669222)

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 3]
```diff
@@ -345,6 +345,32 @@ def load_arguments(self, _):
         c.argument('workspace_name', options_list='--workspace-name')
     # endregion
 
+    # region Log Analytics Workspace Linked Service
+    with self.argument_context('monitor log-analytics workspace linked-service') as c:
+        c.argument('linked_service_name', name_arg_type, help='Name of the linkedServices resource. Supported values: cluster, automation.')
+        c.argument('workspace_name', options_list='--workspace-name')
+        c.argument('resource_id', help='The resource id of the resource that will be linked to the workspace. This '
```
[the remark this answers is not in the document]
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/monitor/_params.py:352]: The resource type is defined by linked_service_name which support `automation` and `cluster`

---
In reply to: [413634067](https://github.com/Azure/azure-cli/pull/13133#discussion_r413634067) [](ancestors = 413634067)

File: src/azure-cli/azure/cli/command_modules/monitor/_params.py [hunk 4]
```diff
@@ -345,6 +345,32 @@ def load_arguments(self, _):
...
+    # endregion
+
+    # region Log Analytics Cluster
+    with self.argument_context('monitor log-analytics cluster') as c:
+        c.argument('cluster_name', name_arg_type, help='The name of the Log Analytics cluster.')
+        c.argument('sku_name', help="The name of the SKU. Currently only support 'CapacityReservation'")
+        c.argument('sku_capacity', help='The capacity of the SKU. It must be in the range of 1000-2000 per day and must'
+                                        ' be in multiples of 100. If you want to increase the limit, please contact'
+                                        ' LAIngestionRate@microsoft.com. It can be decreased only after 31 days.')
+        c.argument('identity_type', help='The identity type. Supported values: SystemAssigned, None')
```
[the remark this answers is not in the document]
Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/monitor/_params.py]: Seem not support 'None' and will confirm with service team

---
In reply to: [413634993](https://github.com/Azure/azure-cli/pull/13133#discussion_r413634993) [](ancestors = 413634993)

**Pull request merged**